core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -32,7 +32,8 @@ private[spark] class ApplicationInfo(
     val desc: ApplicationDescription,
     val submitDate: Date,
     val driver: RpcEndpointRef,
-    defaultCores: Int)
+    defaultCores: Int,
+    var appUiUrl: String)
```

Comment:
This is the only possible downside, that you only moved the mutable field to another class. Still, I think it is a reasonable update. @JoshRosen  ?